src/node/quote_endorsements_client.h

```diff
@@ -0,0 +1,65 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the Apache 2.0 License.
+#pragma once
+
+#include "ccf/pal/attestation.h"
+#include "enclave/rpc_sessions.h"
+
+namespace ccf
+{
+  using QuoteEndorsementsFetchedCallback =
+    std::function<void(std::vector<uint8_t>&& endorsements)>;
+
+  // Resilient client to fetch attestation report endorsement certificate.
+  class QuoteEndorsementsClient
+  {
+  private:
+    std::shared_ptr<RPCSessions> rpcsessions;
+
+  public:
+    QuoteEndorsementsClient(const std::shared_ptr<RPCSessions>& rpcsessions_) :
+      rpcsessions(rpcsessions_){};
+
+    void fetch_endorsements(
+      const pal::EndorsementEndpointConfiguration& config,
+      QuoteEndorsementsFetchedCallback cb)
+    {
+      // Note: server CA is not checked here as this client is not sending
+      // private data. If the server was malicious and the certificate chain was
+      // bogus, the verification of the endorsement of the quote would fail
+      // anyway.
+      auto unauthenticated_client =
+        rpcsessions->create_client(std::make_shared<tls::Cert>(
+          nullptr, std::nullopt, std::nullopt, std::nullopt, false));
+
+      unauthenticated_client->connect(
+        config.host,
+        config.port,
+        [cb](
+          http_status status,
+          http::HeaderMap&& headers,
+          std::vector<uint8_t>&& data) {
+          if (status != HTTP_STATUS_OK)
+          {
+            LOG_FAIL_FMT(
+              "Error fetching endorsements for attestation report: {}", status);
+          }
+
+          cb(std::move(data));
+        },
+        [](const std::string& error_msg) {
+          // TLS errors should be handled here
```

Comment:
Does this need doing in this PR?

Reply:
I'm planning to do this as part of 2. in #4144 in a follow-up PR.